checkov/secrets/scan_git_history.py

```diff
@@ -28,134 +30,149 @@ class EnrichedPotentialSecret(TypedDict):
-                   file_name: str, commit_hash: str) -> None:
-    # First find if secret was moved in the file
-    equal_secret_in_commit: Dict[str, List[str]] = defaultdict(list)
-    for secret in file_results:
-        secret_key = f'{secret.filename}_{secret.secret_hash}_{secret.type.replace(" ", "-")}'
-        equal_secret_in_commit[secret_key].append('added' if secret.is_added else 'removed')
-
-    for secret in file_results:
-        secret_key = f'{secret.filename}_{secret.secret_hash}_{secret.type.replace(" ", "-")}'
-        if all(value in equal_secret_in_commit[secret_key] for value in ['added', 'removed']):
-            continue
-        if secret.is_added:
-            add_new_secret_to_map(secret_key, secret_map, commit_hash, secret)
-        if secret.is_removed:
-            update_removed_secret_in_map(secret_map, commit_hash, secret_key, secret, file_name)
-
-
-def add_new_secret_to_map(secret_key: str,
-                          secret_map: Dict[str, List[EnrichedPotentialSecret]],
-                          commit_hash: str,
-                          secret: PotentialSecret) -> None:
-    if secret_key not in secret_map:
-        secret_map[secret_key] = []
-    else:
-        all_removed = all(
-            potential_secret.get('removed_commit_hash') for potential_secret in secret_map[secret_key])
-        # Update secret map with the new potential secret
-        if all_removed:
-            secret_map[secret_key][0].update({'potential_secret': secret, 'removed_commit_hash': ''})
-            return
-    secret_map[secret_key].append(
-        {'added_commit_hash': commit_hash, 'removed_commit_hash': '', 'potential_secret': secret})
-
-
-def update_removed_secret_in_map(secret_map: Dict[str, List[EnrichedPotentialSecret]],
-                                 commit_hash: str,
-                                 secret_key: str,
-                                 secret: PotentialSecret,
-                                 file_name: str) -> None:
-    # Try to find the corresponding added secret in the git history secret map
-    try:
-        for secret_in_file in secret_map[secret_key]:
-            # Compared by filename, secret_hash, and type
-            if secret.__eq__(secret_in_file['potential_secret']):
+        return commits_diff
+
+
+class GitHistorySecretStore:
+    def __init__(self) -> None:
+        self.secrets_by_file_value_type: Dict[str, List[EnrichedPotentialSecret]] = {}
+
+    def set_secret_map(self, file_results: List[PotentialSecret], file_name: str, commit_hash: str) -> None:
+        # First find if secret was moved in the file
+        equal_secret_in_commit: Dict[str, List[str]] = defaultdict(list)
+        for secret in file_results:
+            secret_key = f'{file_name}_{secret.secret_hash}_{secret.type.replace(" ", "-")}'
+            equal_secret_in_commit[secret_key].append('added' if secret.is_added else 'removed')
```

Comment:
We are not checking is_removed here before setting to removed?

Reply:
yes because every line in the diff is added or removed so 'not added' is 'removed'